src/main/kotlin/org/pkl/lsp/ast/Extensions.kt

```diff
@@ -172,27 +170,25 @@ private fun PklNode.getEscapedText(): String? = buildString {
       TokenType.MLEndQuote -> {} // ignore open/close quotes
       TokenType.SLCharacters,
       TokenType.MLCharacters -> append(terminal.text)
-      TokenType.SLCharacterEscape,
-      TokenType.MLCharacterEscape -> {
-        val text = terminal.text
-        when (text[text.lastIndex]) {
-          'n' -> append('\n')
-          'r' -> append('\r')
-          't' -> append('\t')
-          '\\' -> append('\\')
-          '"' -> append('"')
-          else -> throw AssertionError("Unknown char escape: $text")
-        }
-      }
-      TokenType.SLUnicodeEscape,
-      TokenType.MLUnicodeEscape -> {
+      TokenType.CharacterEscape -> {
         val text = terminal.text
-        val index = text.indexOf('{') + 1
-        if (index != -1) {
-          val hexString = text.substring(index, text.length - 1)
-          try {
-            append(Character.toChars(Integer.parseInt(hexString, 16)))
-          } catch (ignored: NumberFormatException) {} catch (ignored: IllegalArgumentException) {}
+        if (text.contains("u{")) {
+          val index = text.indexOf('{') + 1
+          if (index != -1) {
+            val hexString = text.substring(index, text.length - 1)
+            try {
+              append(Character.toChars(Integer.parseInt(hexString, 16)))
+            } catch (ignored: NumberFormatException) {} catch (ignored: IllegalArgumentException) {}
+          }
+        } else {
```

Comment:
I'm not entirely sure how this worked before, but... `indexOf` returns `-1` when the pattern isn't found. I don't think it ever returns `-2`, does it? I guess it works because `indexOf` _never_ returns `-1`, considering this happens after we know `text.contains("u{")`. I'm also a _little_ thrown by `text.length - 1` for the end of the `hexString`; if we know the string has no trailing matter, why wasn't this a `text.startsWith("u{")` and `val index = 2` (which I'd expect if we believe `text` is "tight")?

Reply:
You are right about the `if (index != -1)`. It won't ever be false because we already made sure with `contains` that this is a unicode escape.
About why `index` is not just `= 2`: we don't know how many `#` chars the string has so the escape may be `\###u{22}`.
Why the `text.length - 1`: because the escape ends with `}` which we have to remove to just get the number.
We could use a simple `\d+` regex to get just the number from the string, but using index is faster.